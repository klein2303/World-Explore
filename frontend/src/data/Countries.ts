import {CountryType} from "../types/CountryTypes";

const Countries: CountryType[] = [
    {
        name: "Egypt",
        continent: "Africa",
        capital: "Cairo",
        largest_city: "Cairo",
        currency: "Egyptian Pound",
        language: "Arabic",
        population: 104258327,
        land_area: 1002450,
        agriculture_area: "3.6%",
        forest_area: "0.1%",
        co2_emissions: 2.4,
        image: "https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg",
    },
    {
        name: "Australia",
        continent: "Oceania",
        capital: "Canberra",
        largest_city: "Sydney",
        currency: "Australian Dollar",
        language: "English",
        population: 25687041,
        land_area: 7692024,
        agriculture_area: "53%",
        forest_area: "16%",
        co2_emissions: 16.9,
        image: "https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg",
    },
    {
        name: "Brazil",
        continent: "South America",
        capital: "Brasília",
        largest_city: "São Paulo",
        currency: "Brazilian Real",
        language: "Portuguese",
        population: 213993437,
        land_area: 8515767,
        agriculture_area: "32%",
        forest_area: "59%",
        co2_emissions: 2.5,
        image: "https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg",
    },
    {
        name: "Nigeria",
        continent: "Africa",
        capital: "Abuja",
        largest_city: "Lagos",
        currency: "Naira",
        language: "English",
        population: 223804632,
        land_area: 923768,
        agriculture_area: "77%",
        forest_area: "9%",
        co2_emissions: 0.7,
        image: "https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg",
    },
    {
        name: "Germany",
        continent: "Europe",
        capital: "Berlin",
        largest_city: "Berlin",
        currency: "Euro",
        language: "German",
        population: 83166711,
        land_area: 357022,
        agriculture_area: "47%",
        forest_area: "33%",
        co2_emissions: 8.4,
        image: "https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg",
    },
    {
        name: "South Africa",
        continent: "Africa",
        capital: "Pretoria (administrative)",
        largest_city: "Johannesburg",
        currency: "South African Rand",
        language: "11 official languages including English, Zulu, and Afrikaans",
        population: 59308690,
        land_area: 1221037,
        agriculture_area: "13%",
        forest_area: "8%",
        co2_emissions: 9.1,
        image: "https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg",
    },
    {
        name: "France",
        continent: "Europe",
        capital: "Paris",
        largest_city: "Paris",
        currency: "Euro",
        language: "French",
        population: 67413000,
        land_area: 551695,
        agriculture_area: "52%",
        forest_area: "31%",
        co2_emissions: 5.5,
        image: "https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg",
    },
    {
        name: "Japan",
        continent: "Asia",
        capital: "Tokyo",
        largest_city: "Tokyo",
        currency: "Japanese Yen",
        language: "Japanese",
        population: 125585000,
        land_area: 377975,
        agriculture_area: "12%",
        forest_area: "67%",
        co2_emissions: 8.9,
        image: "https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg",
    },
    {
        name: "India",
        continent: "Asia",
        capital: "New Delhi",
        largest_city: "Mumbai",
        currency: "Indian Rupee",
        language: "Hindi, English, and 21 other official languages",
        population: 1417173173,
        land_area: 3287263,
        agriculture_area: "60%",
        forest_area: "24%",
        co2_emissions: 1.9,
        image: "https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg",
    },
    {
        name: "United States",
        continent: "North America",
        capital: "Washington, D.C.",
        largest_city: "New York City",
        currency: "United States Dollar",
        language: "English",
        population: 331893745,
        land_area: 9833517,
        agriculture_area: "44%",
        forest_area: "33%",
        co2_emissions: 15.2,
        image: "https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg",
    },
    {
        name: "Canada",
        continent: "North America",
        capital: "Ottawa",
        largest_city: "Toronto",
        currency: "Canadian Dollar",
        language: "English, French",
        population: 38131104,
        land_area: 9984670,
        agriculture_area: "7%",
        forest_area: "34%",
        co2_emissions: 14.7,
        image: "https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg",
    },
    {
        name: "China",
        continent: "Asia",
        capital: "Beijing",
        largest_city: "Shanghai",
        currency: "Renminbi (Yuan)",
        language: "Mandarin",
        population: 1411778724,
        land_area: 9596960,
        agriculture_area: "54%",
        forest_area: "23%",
        co2_emissions: 10.1,
        image: "https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg",
    },
    {
        name: "Argentina",
        continent: "South America",
        capital: "Buenos Aires",
        largest_city: "Buenos Aires",
        currency: "Argentine Peso",
        language: "Spanish",
        population: 45791307,
        land_area: 2780400,
        agriculture_area: "54%",
        forest_area: "10%",
        co2_emissions: 3.9,
        image: "https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg",
    },
    {
        name: "Mexico",
        continent: "North America",
        capital: "Mexico City",
        largest_city: "Mexico City",
        currency: "Mexican Peso",
        language: "Spanish",
        population: 126705138,
        land_area: 1964375,
        agriculture_area: "54%",
        forest_area: "34%",
        co2_emissions: 3.6,
        image: "https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg",
    },
];

export default Countries;
