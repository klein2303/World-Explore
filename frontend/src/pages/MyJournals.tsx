import { useState, useEffect } from "react";
import Navbar from "../components/Navbar/Navbar";
import JournalCard from "../components/JournalCard/JournalCard";
import styles from "../styles/MyJournals.module.css";
import Pagination from "@mui/material/Pagination";
import { useRecoilState } from "recoil";
import { myJournalPageAtom } from "../atoms/MyJournalPageAtom";
import { gql, useQuery } from "@apollo/client";
import { Link } from "react-router-dom";
import { removeQuotes } from "../utils/utils";

const MyJournals = () => {
    const ITEMS_PER_PAGE = 15;
    const [currentPage, setCurrentPage] = useRecoilState(myJournalPageAtom);

    const initialSubtitle =
        window.innerWidth <= 768
            ? "Tap the images to dive into your adventures!"
            : "Your travel stories, captured and cherished forever.";
    const [subtitleText, setSubtitleText] = useState<string>(initialSubtitle);

    const handleChange = (_event: React.ChangeEvent<unknown>, value: number) => {
        setCurrentPage({ page: value });
    };

    const profileid = removeQuotes(sessionStorage.getItem("user")!);

    const WRITTEN_JOURNALS = gql`
        query GetWrittenJournals($skip: Int, $profileid: String!) {
            writtenjournals(skip: $skip, profileid: $profileid) {
                countryid
                countryimage
            }
        }
    `;

    const { data, loading, error } = useQuery(WRITTEN_JOURNALS, {
        variables: { skip: currentPage.page > 0 ? (currentPage.page - 1) * 12 : 0, profileid: profileid },
        fetchPolicy: "cache-first", // Used for first execution
        nextFetchPolicy: "cache-first", // Used for subsequent executions
    });

    // Keep this effect to update the subtitle based on window width
    useEffect(() => {
        const handleResize = () => {
            if (window.innerWidth <= 768) {
                setSubtitleText("Tap the images to dive into your adventures!");
            } else {
                setSubtitleText("Your travel stories, captured and cherished forever.");
            }
        };

        window.addEventListener("resize", handleResize);
        return () => {
            window.removeEventListener("resize", handleResize);
        };
    }, []);

    if (loading) return <p>Loading...</p>;
    if (error) return <p>Error: {error.message}</p>;

    return (
        <>
            <Navbar />
            <main className={styles.container}>
                {/* Title and Tabs in the same row */}
                <header className={styles.header}>
                    <h1 className={styles.pageTitle}>{"Captured Adventures"}</h1>
                </header>
                {data.writtenjournals.length === 0 ? (
                    <>
                        <p className={styles.noResultsMessage}>
                            You havenÂ´t written any journal entries yet. Explore all the countries in the world here:
                            <Link to={"/ExploreCountries"}>Explore Countries Page</Link>
                        </p>
                    </>
                ) : (
                    <>
                        <h2 className={styles.subtitle}>{subtitleText}</h2>
                        <section id="journals-panel" role="tabpanel" className={styles.grid}>
                            {data.writtenjournals.map((journal: { countryid: string; countryimage: string }) => (
                                <JournalCard
                                    key={journal.countryid}
                                    country={journal.countryid}
                                    image={journal.countryimage}
                                />
                            ))}
                        </section>
                        <Pagination
                            page={currentPage.page}
                            onChange={handleChange}
                            count={Math.ceil(data.writtenjournals.length / ITEMS_PER_PAGE)}
                            className={styles.pagination}
                        />
                    </>
                )}
            </main>
        </>
    );
};

export default MyJournals;
