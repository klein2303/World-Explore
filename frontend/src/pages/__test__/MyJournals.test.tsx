import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { expect, it, describe } from 'vitest';
import MyJournals from '../MyJournals';
import Countries from '../../data/Countries';

describe('MyJournals', () => {
  const renderWithRouter = (ui: React.ReactElement) => {
    return render(ui, { wrapper: BrowserRouter });
  };

  it('renders correctly and matches snapshot', () => {
    const { asFragment } = renderWithRouter(<MyJournals />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the correct page title based on the active tab', () => {
    renderWithRouter(<MyJournals />);
    expect(screen.getByText('Captured Adventures')).toBeInTheDocument();

    fireEvent.click(screen.getByText('Unwritten Adventures'));
    expect(screen.getByText('Places Yet to Journal')).toBeInTheDocument();
  });

  it('displays the correct subtitle based on the active tab', () => {
    renderWithRouter(<MyJournals />);
    expect(screen.getByText("Your travel stories, captured and cherished forever.")).toBeInTheDocument();

    fireEvent.click(screen.getByText('Unwritten Adventures'));
    expect(screen.getByText("You’ve visited, but the story’s still untold. Ready to write?")).toBeInTheDocument();
  });

  it('displays the correct content based on the active tab', () => {
    renderWithRouter(<MyJournals />);
    expect(screen.getByLabelText('Journal card for ' + Countries[0].name)).toBeInTheDocument();

    fireEvent.click(screen.getByText('Unwritten Adventures'));
    expect(screen.getByLabelText('Journal card for ' + Countries[10].name)).toBeInTheDocument();
  });

  it('toggles the active tab correctly', () => {
    renderWithRouter(<MyJournals />);
    const journalsTab = screen.getByText('Journal Entries');
    const unwrittenTab = screen.getByText('Unwritten Adventures');

    expect(journalsTab).toHaveAttribute('aria-selected', 'true');
    expect(unwrittenTab).toHaveAttribute('aria-selected', 'false');

    fireEvent.click(unwrittenTab);

    expect(journalsTab).toHaveAttribute('aria-selected', 'false');
    expect(unwrittenTab).toHaveAttribute('aria-selected', 'true');
  });
});