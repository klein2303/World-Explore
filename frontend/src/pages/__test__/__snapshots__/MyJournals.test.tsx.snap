// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyJournals > renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <main>
    <nav
      aria-label="Main Navigation"
      class="_navbar_278c00"
      role="navigation"
    >
      <svg
        aria-label="Open Mobile Menu"
        class="_hamburgmenu_278c00"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 15 15"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M1.5 3C1.22386 3 1 3.22386 1 3.5C1 3.77614 1.22386 4 1.5 4H13.5C13.7761 4 14 3.77614 14 3.5C14 3.22386 13.7761 3 13.5 3H1.5ZM1 7.5C1 7.22386 1.22386 7 1.5 7H13.5C13.7761 7 14 7.22386 14 7.5C14 7.77614 13.7761 8 13.5 8H1.5C1.22386 8 1 7.77614 1 7.5ZM1 11.5C1 11.2239 1.22386 11 1.5 11H13.5C13.7761 11 14 11.2239 14 11.5C14 11.7761 13.7761 12 13.5 12H1.5C1.22386 12 1 11.7761 1 11.5Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <a
        class="_navTitle_278c00"
        href="/"
      >
        <p>
          WorldExplore
        </p>
      </a>
      <div
        class="_navlinks_278c00"
      >
        <a
          class="_navlink_278c00"
          href="/ExploreCountries"
        >
          <svg
            aria-hidden="true"
            class="_search_278c00"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.25 2a8.25 8.25 0 0 1 6.34 13.53l5.69 5.69a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-5.69-5.69A8.25 8.25 0 1 1 10.25 2ZM3.5 10.25a6.75 6.75 0 1 0 13.5 0 6.75 6.75 0 0 0-13.5 0Z"
            />
          </svg>
          <p>
            Explore Countries
          </p>
        </a>
        <a
          class="_navlink_278c00"
          href="/MyJournals"
        >
          <svg
            aria-hidden="true"
            class="_pin_278c00"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 10c0 6-8 12-8 12s-8-6-8-12a8 8 0 0 1 16 0Z"
            />
            <circle
              cx="12"
              cy="10"
              r="3"
            />
          </svg>
          <p>
            My Journals
          </p>
        </a>
        <svg
          aria-label="Open Profile Menu"
          class="_profile_278c00"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M16 9C16 11.2091 14.2091 13 12 13C9.79086 13 8 11.2091 8 9C8 6.79086 9.79086 5 12 5C14.2091 5 16 6.79086 16 9ZM14 9C14 10.1046 13.1046 11 12 11C10.8954 11 10 10.1046 10 9C10 7.89543 10.8954 7 12 7C13.1046 7 14 7.89543 14 9Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1ZM3 12C3 14.0902 3.71255 16.014 4.90798 17.5417C6.55245 15.3889 9.14627 14 12.0645 14C14.9448 14 17.5092 15.3531 19.1565 17.4583C20.313 15.9443 21 14.0524 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12ZM12 21C9.84977 21 7.87565 20.2459 6.32767 18.9878C7.59352 17.1812 9.69106 16 12.0645 16C14.4084 16 16.4833 17.1521 17.7538 18.9209C16.1939 20.2191 14.1881 21 12 21Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </nav>
  </main>
  <main
    class="_container_5b0c23"
  >
    <header
      class="_header_5b0c23"
    >
      <h1
        class="_pageTitle_5b0c23"
      >
        Captured Adventures
      </h1>
      <div
        aria-label="Journal Tabs"
        class="_tabs_5b0c23"
        role="tablist"
      >
        <button
          aria-controls="journals-panel"
          aria-selected="true"
          class="_tabButton_5b0c23 _active_5b0c23"
          id="journals-tab"
          role="tab"
        >
          Journal Entries
        </button>
        <button
          aria-controls="unwritten-panel"
          aria-selected="false"
          class="_tabButton_5b0c23 "
          id="unwritten-tab"
          role="tab"
        >
          Unwritten Adventures
        </button>
      </div>
    </header>
    <p
      class="_subtitle_5b0c23"
    >
      Your travel stories, captured and cherished forever.
    </p>
    <section
      aria-labelledby="journals-tab"
      class="_grid_5b0c23"
      id="journals-panel"
      role="tabpanel"
    >
      <div
        class="_cardWrapper_be814f"
      >
        <section
          aria-label="Journal card for Aountry1"
          class="_card_be814f"
        >
          <div
            class="_verticalStrip_be814f"
          />
          <img
            alt="Image of Aountry1"
            class="_cardImage_be814f"
            src="https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg"
          />
          <div
            class="_overlayText_be814f"
          >
            Read Journal
          </div>
        </section>
        <div
          class="_cardText_be814f"
        >
          <p
            class="_cardTitle_be814f"
          >
            Aountry1
          </p>
          <p
            class="_cardDate_be814f"
          >
            2023-01-12
          </p>
        </div>
      </div>
      <div
        class="_cardWrapper_be814f"
      >
        <section
          aria-label="Journal card for Bountry2"
          class="_card_be814f"
        >
          <div
            class="_verticalStrip_be814f"
          />
          <img
            alt="Image of Bountry2"
            class="_cardImage_be814f"
            src="https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg"
          />
          <div
            class="_overlayText_be814f"
          >
            Read Journal
          </div>
        </section>
        <div
          class="_cardText_be814f"
        >
          <p
            class="_cardTitle_be814f"
          >
            Bountry2
          </p>
          <p
            class="_cardDate_be814f"
          >
            2023-01-12
          </p>
        </div>
      </div>
      <div
        class="_cardWrapper_be814f"
      >
        <section
          aria-label="Journal card for Country3"
          class="_card_be814f"
        >
          <div
            class="_verticalStrip_be814f"
          />
          <img
            alt="Image of Country3"
            class="_cardImage_be814f"
            src="https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg"
          />
          <div
            class="_overlayText_be814f"
          >
            Read Journal
          </div>
        </section>
        <div
          class="_cardText_be814f"
        >
          <p
            class="_cardTitle_be814f"
          >
            Country3
          </p>
          <p
            class="_cardDate_be814f"
          >
            2023-01-12
          </p>
        </div>
      </div>
      <div
        class="_cardWrapper_be814f"
      >
        <section
          aria-label="Journal card for Dountry4"
          class="_card_be814f"
        >
          <div
            class="_verticalStrip_be814f"
          />
          <img
            alt="Image of Dountry4"
            class="_cardImage_be814f"
            src="https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg"
          />
          <div
            class="_overlayText_be814f"
          >
            Read Journal
          </div>
        </section>
        <div
          class="_cardText_be814f"
        >
          <p
            class="_cardTitle_be814f"
          >
            Dountry4
          </p>
          <p
            class="_cardDate_be814f"
          >
            2023-01-12
          </p>
        </div>
      </div>
      <div
        class="_cardWrapper_be814f"
      >
        <section
          aria-label="Journal card for Eountry5"
          class="_card_be814f"
        >
          <div
            class="_verticalStrip_be814f"
          />
          <img
            alt="Image of Eountry5"
            class="_cardImage_be814f"
            src="https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg"
          />
          <div
            class="_overlayText_be814f"
          >
            Read Journal
          </div>
        </section>
        <div
          class="_cardText_be814f"
        >
          <p
            class="_cardTitle_be814f"
          >
            Eountry5
          </p>
          <p
            class="_cardDate_be814f"
          >
            2023-01-12
          </p>
        </div>
      </div>
      <div
        class="_cardWrapper_be814f"
      >
        <section
          aria-label="Journal card for Fountry6"
          class="_card_be814f"
        >
          <div
            class="_verticalStrip_be814f"
          />
          <img
            alt="Image of Fountry6"
            class="_cardImage_be814f"
            src="https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg"
          />
          <div
            class="_overlayText_be814f"
          >
            Read Journal
          </div>
        </section>
        <div
          class="_cardText_be814f"
        >
          <p
            class="_cardTitle_be814f"
          >
            Fountry6
          </p>
          <p
            class="_cardDate_be814f"
          >
            2023-01-12
          </p>
        </div>
      </div>
      <div
        class="_cardWrapper_be814f"
      >
        <section
          aria-label="Journal card for Gountry7"
          class="_card_be814f"
        >
          <div
            class="_verticalStrip_be814f"
          />
          <img
            alt="Image of Gountry7"
            class="_cardImage_be814f"
            src="https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg"
          />
          <div
            class="_overlayText_be814f"
          >
            Read Journal
          </div>
        </section>
        <div
          class="_cardText_be814f"
        >
          <p
            class="_cardTitle_be814f"
          >
            Gountry7
          </p>
          <p
            class="_cardDate_be814f"
          >
            2023-01-12
          </p>
        </div>
      </div>
      <div
        class="_cardWrapper_be814f"
      >
        <section
          aria-label="Journal card for Hountry8"
          class="_card_be814f"
        >
          <div
            class="_verticalStrip_be814f"
          />
          <img
            alt="Image of Hountry8"
            class="_cardImage_be814f"
            src="https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg"
          />
          <div
            class="_overlayText_be814f"
          >
            Read Journal
          </div>
        </section>
        <div
          class="_cardText_be814f"
        >
          <p
            class="_cardTitle_be814f"
          >
            Hountry8
          </p>
          <p
            class="_cardDate_be814f"
          >
            2023-01-12
          </p>
        </div>
      </div>
      <div
        class="_cardWrapper_be814f"
      >
        <section
          aria-label="Journal card for Iountry9"
          class="_card_be814f"
        >
          <div
            class="_verticalStrip_be814f"
          />
          <img
            alt="Image of Iountry9"
            class="_cardImage_be814f"
            src="https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg"
          />
          <div
            class="_overlayText_be814f"
          >
            Read Journal
          </div>
        </section>
        <div
          class="_cardText_be814f"
        >
          <p
            class="_cardTitle_be814f"
          >
            Iountry9
          </p>
          <p
            class="_cardDate_be814f"
          >
            2023-01-12
          </p>
        </div>
      </div>
      <div
        class="_cardWrapper_be814f"
      >
        <section
          aria-label="Journal card for Jountry10"
          class="_card_be814f"
        >
          <div
            class="_verticalStrip_be814f"
          />
          <img
            alt="Image of Jountry10"
            class="_cardImage_be814f"
            src="https://images.pexels.com/photos/346529/pexels-photo-346529.jpeg"
          />
          <div
            class="_overlayText_be814f"
          >
            Read Journal
          </div>
        </section>
        <div
          class="_cardText_be814f"
        >
          <p
            class="_cardTitle_be814f"
          >
            Jountry10
          </p>
          <p
            class="_cardDate_be814f"
          >
            2023-01-12
          </p>
        </div>
      </div>
    </section>
    <section
      aria-labelledby="unwritten-tab"
      class="_grid_5b0c23"
      hidden=""
      id="unwritten-panel"
      role="tabpanel"
    />
  </main>
</DocumentFragment>
`;
